cmd/daemon/main.go

```diff
@@ -397,6 +406,25 @@ func loadConfig(cfg *Config) error {
 		return err
 	}
 
+	// Make config directory if needed.
+	err = os.MkdirAll(cfg.ConfigDir, 0o700)
+	if err != nil {
+		return fmt.Errorf("failed creating config directory: %w", err)
+	}
+
+	// Lock the config directory (to ensure multiple instances won't clobber
+	// each others state).
+	lockFilePath := filepath.Join(cfg.ConfigDir, "lockfile")
+	cfg.configLock = flock.New(lockFilePath)
+	locked, err := cfg.configLock.TryLock()
+	if err != nil {
+		return fmt.Errorf("could not lock config directory: %w", err)
+	}
+	if !locked {
+		// Lock already taken! Looks like go-librespot is already running.
+		return fmt.Errorf("%w (lockfile: %s)", errAlreadyRunning, lockFilePath)
+	}
+
```

Comment:
This could be compacted to a single block so that `locked` does not clobber. Just nitpicking...

Reply:
You mean something like this?

```go
	if locked, err := cfg.configLock.TryLock(); err != nil {
		return fmt.Errorf("could not lock config directory: %w", err)
	} else if !locked {
		// Lock already taken! Looks like go-librespot is already running.
		return fmt.Errorf("%w (lockfile: %s)", errAlreadyRunning, lockFilePath)
	}
```

I guess, yeah. I'm not sure it's more readable but I'm fine either way.

Reply:
Yeah... Just not to have `locked` in the scope

Reply:
Fixed!